make test on darwin fails with error "build constraints exclude all Go files"

### What happened?

Tried running make test against local latest fork of kubernetes master

```
%  make test
+++ [0402 17:43:52] Set GOMAXPROCS automatically to 8
WARNING: ulimit -n (files) should be at least 1000, is 256, may cause test failure
+++ [0402 17:47:36] Running tests without code coverage and with -race
package k8s.io/kubernetes/pkg/proxy/iptables: build constraints exclude all Go files in /Users/karthikkn/karthik-k8-workspace/kubernetes/pkg/proxy/iptables
package k8s.io/kubernetes/pkg/proxy/ipvs/ipset: build constraints exclude all Go files in /Users/karthikkn/karthik-k8-workspace/kubernetes/pkg/proxy/ipvs/ipset
package k8s.io/kubernetes/pkg/proxy/ipvs/ipset/testing: build constraints exclude all Go files in /Users/karthikkn/karthik-k8-workspace/kubernetes/pkg/proxy/ipvs/ipset/testing
package k8s.io/kubernetes/pkg/proxy/ipvs/testing: build constraints exclude all Go files in /Users/karthikkn/karthik-k8-workspace/kubernetes/pkg/proxy/ipvs/testing
package k8s.io/kubernetes/pkg/proxy/ipvs/util: build constraints exclude all Go files in /Users/karthikkn/karthik-k8-workspace/kubernetes/pkg/proxy/ipvs/util
package k8s.io/kubernetes/pkg/proxy/ipvs/util/testing: build constraints exclude all Go files in /Users/karthikkn/karthik-k8-workspace/kubernetes/pkg/proxy/ipvs/util/testing
package k8s.io/kubernetes/pkg/proxy/nftables: build constraints exclude all Go files in /Users/karthikkn/karthik-k8-workspace/kubernetes/pkg/proxy/nftables
package k8s.io/kubernetes/pkg/util/iptables/testing: build constraints exclude all Go files in /Users/karthikkn/karthik-k8-workspace/kubernetes/pkg/util/iptables/testing
make: *** [test] Error 1

```

Go version
```
% go version
go version go1.22.0 darwin/amd64
```

Go env

```
% go env
GO111MODULE='on'
GOARCH='amd64'
GOBIN=''
GOCACHE='/Users/karthikkn/Library/Caches/go-build'
GOENV='/Users/karthikkn/Library/Application Support/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='darwin'
GOINSECURE=''
GOMODCACHE='/Users/karthikkn/go/pkg/mod'
GONOPROXY='github.com/PDeXchange'
GONOSUMDB='github.com/PDeXchange'
GOOS='darwin'
GOPATH='/Users/karthikkn/go'
GOPRIVATE='github.com/PDeXchange'
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/Users/karthikkn/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-amd64'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='auto'
GOTOOLDIR='/Users/karthikkn/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.darwin-amd64/pkg/tool/darwin_amd64'
GOVCS=''
GOVERSION='go1.22.0'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='clang'
CXX='clang++'
CGO_ENABLED='1'
GOMOD='/Users/karthikkn/karthik-k8-workspace/kubernetes/go.mod'
GOWORK='/Users/karthikkn/karthik-k8-workspace/kubernetes/go.work'
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -arch x86_64 -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -ffile-prefix-map=/var/folders/j0/f79y3_5d5svcgbm4_yfjg91w0000gn/T/go-build1055166348=/tmp/go-build -gno-record-gcc-switches -fno-common'
```

### What did you expect to happen?

Successfully able to run the test

### How can we reproduce it (as minimally and precisely as possible)?

Try running make test on darwin os.
```
make test
```

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>
Latest master

```
% git log --oneline
e3bb757129c (HEAD -> master, origin/master, origin/HEAD) Merge pull request #124143 from enj/enj/i/svm_update_crd_flake
f338ab5f959 svm test: wait after updating CRD to avoid flakes
79c61d5f030 Merge pull request #124124 from carlory/fix-124120
363fee59e4b fix panic with SIGSEGV in kubeadm certs check-expiration
3dedb8eb8c1 Merge pull request #124111 from liggitt/deflake-authz-test
```

### Cloud provider

<details>

</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
Darwin Karthiks-MacBook-Pro.local 23.4.0 Darwin Kernel Version 23.4.0: Wed Feb 21 21:44:31 PST 2024; root:xnu-10063.101.15~2/RELEASE_X86_64 x86_64

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

--- a/pkg/proxy/iptables/doc.go
+++ b/pkg/proxy/iptables/doc.go
@@ -1,0 +1,19 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is an empty file to avoid "build constraints exclude all Go files" error while running unit tests on non-linux platforms.
+
+package iptables

--- a/pkg/proxy/ipvs/ipset/doc.go
+++ b/pkg/proxy/ipvs/ipset/doc.go
@@ -1,0 +1,19 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is an empty file to avoid "build constraints exclude all Go files" error while running unit tests on non-linux platforms.
+
+package ipset

--- a/pkg/proxy/ipvs/ipset/testing/doc.go
+++ b/pkg/proxy/ipvs/ipset/testing/doc.go
@@ -1,0 +1,19 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is an empty file to avoid "build constraints exclude all Go files" error while running unit tests on non-linux platforms.
+
+package testing

--- a/pkg/proxy/ipvs/testing/doc.go
+++ b/pkg/proxy/ipvs/testing/doc.go
@@ -1,0 +1,19 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is an empty file to avoid "build constraints exclude all Go files" error while running unit tests on non-linux platforms.
+
+package testing

--- a/pkg/proxy/ipvs/util/doc.go
+++ b/pkg/proxy/ipvs/util/doc.go
@@ -1,0 +1,19 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is an empty file to avoid "build constraints exclude all Go files" error while running unit tests on non-linux platforms.
+
+package ipvs

--- a/pkg/proxy/ipvs/util/testing/doc.go
+++ b/pkg/proxy/ipvs/util/testing/doc.go
@@ -1,0 +1,19 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is an empty file to avoid "build constraints exclude all Go files" error while running unit tests on non-linux platforms.
+
+package testing

--- a/pkg/proxy/nftables/doc.go
+++ b/pkg/proxy/nftables/doc.go
@@ -1,0 +1,19 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is an empty file to avoid "build constraints exclude all Go files" error while running unit tests on non-linux platforms.
+
+package nftables

--- a/pkg/util/iptables/testing/doc.go
+++ b/pkg/util/iptables/testing/doc.go
@@ -1,0 +1,19 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is an empty file to avoid "build constraints exclude all Go files" error while running unit tests on non-linux platforms.
+
+package testing

